File: crossecore-java/src/java_/PrefixExpressionBase.java
package java_;
import org.eclipse.emf.common.notify.Notification;
import org.eclipse.emf.common.notify.NotificationChain;
import org.eclipse.emf.common.notify.impl.NotificationImpl;
import org.eclipse.emf.ecore.InternalEObject;
import org.eclipse.emf.ecore.*;
import org.eclipse.emf.ecore.impl.ENotificationImpl;
public class PrefixExpressionBase 
extends ExpressionImpl implements PrefixExpression
{
	protected static final PrefixExpressionKind OPERATOR_EDEFAULT = PrefixExpressionKind.INCREMENT;
	private PrefixExpressionKind _operator = OPERATOR_EDEFAULT;
	public PrefixExpressionKind getOperator()
	{
		return _operator;
	}
	public void setOperator(PrefixExpressionKind value){
		
		PrefixExpressionKind oldValue = _operator;
		_operator = value;
		if (eNotificationRequired())
			eNotify(new ENotificationImpl(this, Notification.SET, Java_PackageImpl.PREFIX_EXPRESSION__OPERATOR, oldValue, value));
		
	}
	
	
	
	private Expression _operand;
	public Expression getOperand()
	{
		
			return _operand;
		
		
	}
	public void setOperand(Expression value){
		if (value != _operand) {
			NotificationChain msgs = null;
			if (_operand != null){
				msgs = ((InternalEObject)_operand).eInverseRemove(this, EOPPOSITE_FEATURE_BASE - Java_PackageImpl.PREFIX_EXPRESSION__OPERAND, null, msgs);
			}
			if (value != null){
				msgs = ((InternalEObject)value).eInverseAdd(this, EOPPOSITE_FEATURE_BASE - Java_PackageImpl.PREFIX_EXPRESSION__OPERAND, null, msgs);
			}
			msgs = basicSetOperand(value, msgs);
			if (msgs != null) {
				msgs.dispatch();
			}
		}
		else if (eNotificationRequired()){
			eNotify(new ENotificationImpl(this, NotificationImpl.SET,Java_PackageImpl.PREFIX_EXPRESSION__OPERAND , value, value));
		}
	}

	
	@Override
	protected EClass eStaticClass() {
		return Java_PackageImpl.Literals.PREFIXEXPRESSION;
	}
	
	
	public NotificationChain basicSetOperand(Expression newobj, NotificationChain msgs) {
		Expression oldobj = _operand;
		_operand = newobj;
		if (eNotificationRequired()) {
			ENotificationImpl notification = new ENotificationImpl(this, NotificationImpl.SET, Java_PackageImpl.PREFIX_EXPRESSION__OPERAND, oldobj, newobj);
			if (msgs == null){
				msgs = notification;
			}
			else{
				msgs.add(notification);
			}
		}
		return msgs;
	}
	
	@Override
	public Object eGet(int featureID, boolean resolve, boolean coreType) {
		switch (featureID) {
			case Java_PackageImpl.PREFIX_EXPRESSION__OPERATOR:
				return getOperator();
			case Java_PackageImpl.PREFIX_EXPRESSION__OPERAND:
				return getOperand();
		}
		return super.eGet(featureID, resolve, coreType);
	}
	
	@Override
	public void eSet(int featureID, Object newValue) {
		switch (featureID) {
			case Java_PackageImpl.PREFIX_EXPRESSION__OPERATOR:
				setOperator((PrefixExpressionKind) newValue);
				return;
			case Java_PackageImpl.PREFIX_EXPRESSION__OPERAND:
				setOperand((Expression) newValue);
				return;
		}
		super.eSet(featureID, newValue);
	}
	
	@Override
	public boolean eIsSet(int featureID) {
		switch (featureID) {
			case Java_PackageImpl.PREFIX_EXPRESSION__OPERATOR:
				return OPERATOR_EDEFAULT == null ? _operator != null : !OPERATOR_EDEFAULT.equals(_operator);
			case Java_PackageImpl.PREFIX_EXPRESSION__OPERAND:
				return _operand != null; //single != null;
		}
		return super.eIsSet(featureID);
	}
	
	
	
}

File: crossecore-csharp/Java/Java/Java_/PrefixExpressionBase.cs
/* CrossEcore is a cross-platform modeling framework that generates C#, TypeScript, 
 * JavaScript, Swift code from Ecore models with embedded OCL (http://www.crossecore.org/).
 * The original Eclipse Modeling Framework is available at https://www.eclipse.org/modeling/emf/.
 * 
 * contributor: Simon Schwichtenberg
 */
 
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using oclstdlib;
using Ecore;
namespace Java_{
	public class PrefixExpressionBase 
	:ExpressionImpl, PrefixExpression
	{
		private PrefixExpressionKind _operator = PrefixExpressionKind.INCREMENT;
		public virtual PrefixExpressionKind operator_
		{
		get { 
			return _operator;
		}
		set { _operator = value; }
		}
		private Expression _operand;
		public virtual Expression operand
		{
			get {
			
				return _operand;
			}
			set {
				if (value != _operand) {
					NotificationChain msgs = null;
					if (_operand != null){
						msgs = ((InternalEObject)_operand).eInverseRemove(this, EOPPOSITE_FEATURE_BASE - Java_PackageImpl.PREFIXEXPRESSION_OPERAND, null, msgs);
					}
					if (value != null){
						msgs = ((InternalEObject)value).eInverseAdd(this, EOPPOSITE_FEATURE_BASE - Java_PackageImpl.PREFIXEXPRESSION_OPERAND, null, msgs);
					}
					msgs = basicSetOperand(value, msgs);
					if (msgs != null) {
						msgs.dispatch();
					}
				}
				else if (eNotificationRequired()){
					eNotify(new ENotificationImpl(this, NotificationImpl.SET,Java_PackageImpl.PREFIXEXPRESSION_OPERAND , value, value));
				}
				}
		}
		
		protected override EClass eStaticClass() {
			return Java_PackageImpl.Literals.PREFIXEXPRESSION;
		}
		
		
		public NotificationChain basicSetOperand(Expression newobj, NotificationChain msgs) {
			var oldobj = _operand;
			_operand = newobj;
			if (eNotificationRequired()) {
				var notification = new ENotificationImpl(this, NotificationImpl.SET, Java_PackageImpl.PREFIXEXPRESSION_OPERAND, oldobj, newobj);
				if (msgs == null){
					msgs = notification;
				}
				else{
					msgs.add(notification);
				}
			}
			return msgs;
		}
		
		public override object eGet(int featureID, bool resolve, bool coreType) {
			switch (featureID) {
				case Java_PackageImpl.PREFIXEXPRESSION_OPERATOR:
					return operator_;
				case Java_PackageImpl.PREFIXEXPRESSION_OPERAND:
					return operand;
			}
			return base.eGet(featureID, resolve, coreType);
		}
		
		
		public override void eSet(int featureID, object newValue) {
			switch (featureID) {
				case Java_PackageImpl.PREFIXEXPRESSION_OPERATOR:
					operator_ = (PrefixExpressionKind) newValue;
					return;
				case Java_PackageImpl.PREFIXEXPRESSION_OPERAND:
					operand = (Expression) newValue;
					return;
			}
			base.eSet(featureID, newValue);
		}
		
		/*
		public override bool eIsSet(int featureID) {
			switch (featureID) {
				case Java_PackageImpl.PREFIXEXPRESSION_OPERATOR:
					return OPERATOR_EDEFAULT == null ? _operator != null : !OPERATOR_EDEFAULT.equals(_operator);
				case Java_PackageImpl.PREFIXEXPRESSION_OPERAND:
					return _operand != null; //single != null;
			}
			return base.eIsSet(featureID);
		}
		*/
		
		
		
	}
}
